source/getting-started/core-concepts.md

```diff
@@ -1,58 +1,136 @@
+is a peice of logic that returns a single value (much like a property
+has only a single value).
+
+For example, this helper and accompianing usage create a full name:
+
+```js
+// app/helpers/full-name.js
+import Ember from 'ember';
+
+export default Ember.Helper.helper(function([firstName, lastName]) {
+  return `Mrs. ${firstName} ${secondName}`;
+});
+```
+
+```handlebars
+{{full-name firstName "Doe"}}
+```
+
+You can read more about helpers in [this API documentation](http://emberjs.com/api/classes/Ember.Helper.html).
 
 ## Components
 
 Components are the primary way user interfaces are organized in Ember. They
-consist of two parts: a template, and a source file written in JavaScript that
-defines the component's behavior. For example, a blog application might have a
-component for displaying a list of blog posts called `all-posts`, and another
-component for displaying an individual post called `view-post`. If users can
-upvote a post, the `view-post` component might define a behavior like _when the
-user clicks the upvote button, increase the `vote` property's value by 1_.
+consist of two parts:
 
-## Controllers
+* A template, for example `app/templates/components/welcome-message.hbs`
+* Optionally a source file written in JavaScript, for example
+  `app/components/welcome-message.js`.
+* **Components must contain a - in their name**. More about that later.
+
+Components differ from helpers in that they can output not only a value, but
+also manage DOM. An important property of components is that they are **isolated**.
+For example, a component's DOM is called "shadow" DOM.
+
+In this example, a component's template adds a div around some other DOM:
 
-Controllers are very much like components, so much so that in future versions of
-Ember, controllers will be replaced entirely with components. At the moment,
-components cannot be routed to (see below), but when this changes, it will be
-recommended to replace all controllers with components.
+```handlebars
+{{! app/templates/components/my-component.hbs }}
+<div class="inner">
+  {{yield}}
+</div>
+```
+
+```handlebars
+<span>Hello</span>
+<div class="outer">
+  {{#my-component}}
+    Hello there!
+  {{/my-component}}
+</div>
+```
+
+Because components deal with DOM, they are also responsible for maintaining
+the bulk of an application's state. Additionally they handle events like
+clicks, keyUp, and other more semtantic actions.
+
+## The Router
+
+The router maps a URL to a template. For example, when a user visits the `/posts`
+URL, the router might load the `posts` route. The router can also load
+nested routes. For example, if a blogging app had a list of blog posts on the
+left of the screen and showed the current blog post on the right, we'd say
+that the `posts/show` route was nested inside the `posts` route.
+
+Perhaps the most important thing to remember about Ember is that **the URL drives
+the application state**. URLs determines what template and route to load, and
```

Comment:
Yup- this section come right after components. The motivation for the ordering in general is two parts:

* How does Ember manage creating HTML and UI?
  * Templates, properties are values
  * Helpers, which are just functions generating values
  * Components, which organize complex API and generate DOM
* How does Ember load a page?
  * The Router handles URLs, loading templates
  * Models provide data to a template
  * Controllers maintain local state for a URL
  * Routes customize the data loading process

Convention over configuration allows us to push controllers and routes to the end, since those things only need to be created if you want to customize their behavior. You could write a complex idiomatic app without ever touching them.